### PR TITLE
generalized version of printgrid

### PR DESCRIPTION
**Pull request by april301975**
the light at the end of the tunnel? :-)

### CODE REVIEW

**Comment by april301975** [comment]
please review my fizzbuzz and series

**Comment by april301975** [comment]
Sorry Natasha... github still confusing to use for me. I reopened it.

**Comment by april301975** [comment]
reopened

File: students/Thai_H/lesson2/PrintGrid.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+def sides(col,size):
+    for row_of_bar in range(size):
+        for c in range(col):
+            print ('|' , end = '')
+            print ((2*size+1) * (' '), end = '')
+            if c == col-1:
+                print ('|')
+
+
+
```
Comment (charlesrobison) [inline students/Thai_H/lesson2/PrintGrid.py]: You'll want to use blank lines sparingly - too many here.  Keep blank lines to 2 between functions, and only one between methods within classes.

File: students/Thai_H/lesson2/PrintGrid.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+def printgrid(row,col,size):
```
Comment (charlesrobison) [inline students/Thai_H/lesson2/PrintGrid.py]: Make sure function names are snake_case.  Also, add spaces after commas: def print_grid(row, col, size):

File: students/Thai_H/lesson2/PrintGrid.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+def top_or_bottom(col,size):
```
Comment (charlesrobison) [inline students/Thai_H/lesson2/PrintGrid.py]: Add spaces after comma: col, size

File: students/Thai_H/lesson2/PrintGrid.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+    for c in range(col):
+        print('+', end = '')
+        print( size * (' -'), end = ' ' )
+        if c == col-1:
+            print ('+')
+
+
+def sides(col,size):
```
Comment (charlesrobison) [inline students/Thai_H/lesson2/PrintGrid.py]: Space here too.

File: students/Thai_H/lesson2/PrintGrid.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        top_or_bottom(col,size)
+        sides(col,size)
+        if r == row-1:
+            top_or_bottom(col,size)
+
+
+##---- print grids of squares
+# (row, column, number of chars as size of edge)
+# all arguments must be int > zero
+printgrid(3, 3, 3)
```
Comment (charlesrobison) [inline students/Thai_H/lesson2/PrintGrid.py]: Function works well!

File: students/Thai_H/lesson2/fizzbuzz.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+for n in range(1, 101):
+    if n % 3 == 0:
+        print('Fizz')
+    elif n % 5 == 0:
+        print('Buzz')
+    elif n % 3 == 0 and n % 5 == 0:
+        print('FizzBuzz')
+    else:
+        print(n)
```
Comment (charlesrobison) [inline students/Thai_H/lesson2/fizzbuzz.py]: Not getting any FizzBuzzes here - take a look at a simple way to rearrange this to get the desired result.

File: students/Thai_H/lesson2/series.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        else:
+            return sum_series(n_index - 1, zero_index = z_idx, one_index = one_idx) + sum_series(n_index-2, zero_index = z_idx, one_index = one_idx)
+
+    #-----------------------
+
+
+#print (fibonacci(6))
+#print (lucas(6))
+print (sum_series(6, zero_index=3, one_index=1))
+print (sum_series.__doc__)
```
Comment (charlesrobison) [inline students/Thai_H/lesson2/series.py]: Missing assert statements - see assignment: "Add a block of code to the end of your series.py module. Use the block to write a series of assert statements that demonstrate that your three functions work properly.

Use comments in this block to inform the observer what your tests do."

**Pull request merged**